src/Authoring/WinRT.SourceGenerator/Helper.cs

```diff
@@ -225,6 +226,40 @@ public static bool IsWinRTType(ISymbol type, Func<ISymbol, bool> isAuthoringWinR
             return isProjectedType;
         }
 
+        public static bool IsWinRTType(ISymbol type, ITypeSymbol winrtRuntimeTypeAttribute, bool isComponentProject, IAssemblySymbol currentAssembly)
+        {
+            if (IsFundamentalType(type))
+            {
+                return true;
+            }
+
+            if (isComponentProject &&
+                // Make sure type is in component project.
+                SymbolEqualityComparer.Default.Equals(type.ContainingAssembly, currentAssembly) &&
+                type.DeclaredAccessibility == Accessibility.Public)
+            {
+                // Authoring diagnostics will make sure all public types are valid WinRT types.
+                return true;
+            }
+
+            bool isProjectedType = HasAttributeWithType(type, winrtRuntimeTypeAttribute);
+            if (!isProjectedType & type.ContainingNamespace != null)
+            {
+                isProjectedType = MappedCSharpTypes.ContainsKey(string.Join(".", type.ContainingNamespace.ToDisplayString(), type.MetadataName));

```

Comment:
Not sure this will work. Doesn't `ContainingNamespace` only return the closest enclosing namespace? Ie. if we have some types with nested namespaces (eg. "System.ComponentModel.DataErrorsChangedEventArgs"), wouldn't that just return the closest namespace name (eg. "ComponentModel")? 🤔

Reply:
It does get the closest namespace, but `ToDisplayString` makes it print out the entire namespace including parent namespaces.  Did a quick test and that seems to confirm that.